src/main/scala/Optics.scala

```diff
@@ -0,0 +1,204 @@
+      new Fold[S, (A, B)] {
+        def foldMap[M: Monoid](f: ((A, B)) => M)(s: S): M =
+          fl.getAll(s).zip(other.getAll(s)).foldMap(f)
+      }
+
+    def flatMap[B](f: A => Fold[S, B]): Fold[S, B] =
+      new Fold[S, B] {
+        def foldMap[M: Monoid](g: B => M)(s: S): M =
+          fl.getAll(s).foldMap(f(_).foldMap(g)(s))
+      }
+  }
+
+  implicit def foldMonad[S] = new Monad[Fold[S, ?]] {
+
+    def point[A](a: => A): Fold[S, A] = new Fold[S, A] {
+      def foldMap[M: Monoid](f: A => M)(s: S): M = f(a)
+    }
+
+    def bind[A, B](fa: Fold[S, A])(f: A => Fold[S, B]): Fold[S, B] =
+      fa.flatMap(f)
+  }
+
+  object Primitives {
+
+    val coupledPeople: Traversal[Couples, Person] = 
+      all ^|->> both
+
+    val people: Traversal[People, Person] = 
+      each[People, Person]
+    
+    val getPeople: People => People = 
+      people.getAll
+
+    val getPeopleName: People => List[String] =
+      (people ^|-> name).getAll
+
+    val getPeopleOnTheirThirties: People => People =
+      people.filter(p => 30 <= p.age && p.age < 40)
+
+    val getHerAges: Couples => List[Int] =
+      (all ^|-> her ^|-> age).getAll
+  }
+
+  object QueryViaQuotation {
+    
+    val differenceFl: Fold[Couples, (String, Int)] =
+      (all ^|-> (him ^|-> age) * (her ^|-> name * age) ^<-> flat).asFold
+        .withFilter { case (ma, _, wa) => wa > ma } 
+        .withMap { case (ma, wn, wa) => (wn, wa - ma) }
+
+    val difference: Couples => List[(String, Int)] = differenceFl.getAll
+
+    difference(couples)
+  }
+
+  object AbstractingOverValues {
+    import Primitives.coupledPeople
+
+    val nameAgeTr = (coupledPeople ^|-> name * age).asFold
+
+    val rangeFl: (Int, Int) => Fold[Couples, String] = { (a, b) =>
+      nameAgeTr.withFilter { case (_, i) => a <= i && i < b }
+               .composeLens(fst)
+    }
+
+    val range: (Int, Int) => Couples => List[String] = { (a, b) => 
+      rangeFl(a, b).getAll 
+    }
+
+    range(30, 40)(couples)
+  }
+
+  object AbstractingOverAPredicate {
+    import AbstractingOverValues.nameAgeTr
+
+    val satisfiesFl: (Int => Boolean) => Fold[Couples, String] = { p =>
+      nameAgeTr.withFilter { case (_, a) => p(a) }
+               .composeLens(fst)
+    }
+
+    val satisfies: (Int => Boolean) => Couples => List[String] = { p =>
+      satisfiesFl(p).getAll
+    }
+
+    satisfies(i => 30 <= i && i < 40)(couples)
+  }
+
+  object ComposingQueries {
+    import AbstractingOverValues.rangeFl
+    
+    def getAgeFl(s: String): Fold[Couples, Int] =
+      all.composeTraversal(both)
+         .asFold
+         .withFilter(_.name == s)
+         .composeLens(age)
+
+    val composeFl = { (s: String, t: String) =>
+      (getAgeFl(s) * getAgeFl(t)).flatMap { case (a, b) => rangeFl(a, b) }
+    }
+
+    // or alternatively
+    val composeFl2 = { (s: String, t: String) =>
+      for {
+        a <- getAgeFl(s)
+        b <- getAgeFl(t)
+        x <- rangeFl(a, b)
+      } yield x
+    }
```

Comment:
`Fold`s as `Monad`s! Thereby, we can implement the *compose* example using a for-comprehension (or horizontal composition as shown in the previous method). I have to check this, but I think that this is given by the fact that this is hiding the List-Kleisli indeed.

\cc @jserranohidalgo